Hash-Tables-and-Hashing Functions.md

```diff
@@ -0,0 +1,150 @@
+# Hash Tables and Hashing Functions
+
+## Introduction to hashing
+
+Hashing is designed to solve the problem of needing to efficiently find or store an item in a collection. 
```

Comment:
I meant to say link to some Wikipedia article that defines Hashing. where you first use the word _Hashing_.